src/Microsoft.ML.OnnxTransform/OnnxTransform.cs

```diff
@@ -208,96 +260,230 @@ public Mapper(IHostEnvironment env, OnnxTransform parent, Schema inputSchema)
                 _host.CheckValue(parent, nameof(parent));
 
                 _parent = parent;
+                _inputColIndices = new int[_parent.Inputs.Length];
+                _isInputVector = new bool[_parent.Inputs.Length];
+                _inputTensorShapes = new OnnxShape[_parent.Inputs.Length];
+                _inputOnnxTypes = new DataType[_parent.Inputs.Length];
```

Comment:
does it make sense to bundle all these into a InputMetadata class and just have one array of that class instead of multiple arrays?

Reply:
This is a matter of readability -- bundling will introduce another data structure. Also, this pattern is what's there in the TF transform -- later on, if we need to consolidate into a single transform which does inferencing of all DNN models, it would be beneficial to keep it this way.

Example here :  [https://github.com/dotnet/machinelearning/blob/cb9effcd091c60fa291aad96cc18c14ddf841b6f/src/Microsoft.ML.TensorFlow/TensorflowTransform.cs#L810](https://github.com/dotnet/machinelearning/blob/cb9effcd091c60fa291aad96cc18c14ddf841b6f/src/Microsoft.ML.TensorFlow/TensorflowTransform.cs#L810) 

---
In reply to: [232838745](https://github.com/dotnet/machinelearning/pull/1586#discussion_r232838745) [](ancestors = 232838745)

Reply:
parallel arrays are considered an anti-pattern in OOD. Unfortunately, they're used often in our codebase. So, up to you if you want to keep it as is or refactor.

[https://codeblog.jonskeet.uk/2014/06/03/anti-pattern-parallel-collections/](https://codeblog.jonskeet.uk/2014/06/03/anti-pattern-parallel-collections/)


---
In reply to: [234077070](https://github.com/dotnet/machinelearning/pull/1586#discussion_r234077070) [](ancestors = 234077070,232838745)